modules/system/classes/MixAssets.php

```diff
@@ -0,0 +1,160 @@
+<?php namespace System\Classes;
+
+use File;
+use ApplicationException;
+use Cms\Classes\Theme;
+use Winter\Storm\Filesystem\PathResolver;
+
+/**
+ * Mix assets using Laravel Mix for Node.js compilation and processing.
+ *
+ * This works similar to the `System\Classes\CombineAssets` class in that it allows modules, plugins and themes to
+ * register configurations that will be passed on to Laravel Mix and Node.js for compilation and processing.
+ *
+ * @package winter\wn-system-module
+ * @author Ben Thomson <git@alfreido.com>, Jack Wilkinson <jax@jaxwilko.com>
+ * @author Winter CMS
+ */
+class MixAssets
+{
+    use \Winter\Storm\Support\Traits\Singleton;
+
+    /**
+     * The filename that stores the package definition.
+     *
+     * @var string
+     */
+    protected $packageJson = 'package.json';
+
+    /**
+     * A list of packages registered for mixing.
+     *
+     * @var array
+     */
+    protected $packages = [];
+
+    /**
+     * Registered callbacks.
+     *
+     * @var array
+     */
+    protected static $callbacks = [];
+
+    /**
+     * Constructor.
+     *
+     * @return void
+     */
+    public function init()
+    {
+        // Call plugins
+        $packages = PluginManager::instance()->getRegistrationMethodValues('registerMixPackages');
+        if (count($packages)) {
+        }
+
+        // Allow current theme to define mix assets
+        $theme = Theme::getActiveTheme();
+
+        if (!is_null($theme)) {
+            $mix = $theme->getConfigValue('mix', []);
+
+            if (count($mix)) {
+                foreach ($mix as $name => $file) {
+                    $path = PathResolver::resolve($theme->getPath() . '/' . $file);
+                    $pinfo = pathinfo($path);
+
+                    $this->registerPackage($name, $pinfo['dirname'], $pinfo['basename']);
+                }
+            }
+        }
```

Comment:
What does a theme definition look like? If it supports an array of mix files should it also support a single mix file as a string?

Reply:
It has to be an array so that the "name" of the package can be defined.

I'm using this for the Workshop theme already - the config looks like this:

```yaml
mix:
    workshop-theme: winter-mix.js
```